src/inertia.js

```diff
@@ -151,23 +151,23 @@ export default {
     }
   },
 
-  replace(url, options = {}) {
+  replace(url, options = { preserveState: true }) {
     return this.visit(url, { ...options, replace: true })
   },
 
   reload(options = {}) {
     return this.replace(window.location.href, options)
   },
 
-  post(url, data = {}, options = {}) {
+  post(url, data = {}, options = { preserveState: true }) {
```

Comment:
Correct me if I'm wrong, but I think if you'd do `Inertia.post(url, data, { preserveScroll: false })`, you'd accidentally set `preserveState` to `false` because that's the default in `setPage`.

Reply:
I don't think so? `post()` calls `visit()`, which internally calls `setPage`.